### DISCUSSION TITLE
Why are some packages automatically updated and others in the same situation are deleted by wingetbot?

### DISCUSSION THREAD

**Discussion by borque** [microsoft/winget-pkgs#82537] [Q&A]
Hello!
Some programmes do not change their installers name when updating but simply replace it with a new version.
This causes a hash-missmatch noticed by wingebot.
As a result, wingebot updates some packages automatically (eg Softmaker Freeoffice with automatic pull request #82503), while others with perfectly the same setting are scheduled for deletion (eg Softmaker Office with automatic pull request #82504).
How can I make wingetbot updating the second package also automatically instead of deleting it, as there is no difference, both simply received an update.
Thanks!

**Comment by mdanish-kh** [comment]
This is because the package being updated is actually replacing the previous manifest/version. They both share the same `InstallerUrl`, so you wouldn't want to have two manifests with same `InstallerUrl` rather only the one which has the correct `InstallerSha256` and the correct `PackageVersion`.

This can be done in a single Update & Remove PR like in this example here https://github.com/microsoft/winget-pkgs/pull/82436 , but wingetbot decides to do it in two separate PRs - one for updating the manifest with new PackageVersion and correct hash, and the other for deleting the previous outdated/incorrect manifest.

In the case a manifest is automatically updated without a deletion PR, it's because the `PackageVersion` remains the same in the update which does not require the creation of a new directory and deletion of the previous one. It just needs an update within the same directory so there is no deletion PR for it. Example being this PR https://github.com/microsoft/winget-pkgs/pull/82492
